### PR TITLE
Date input change workflow

### PR DESCRIPTION
**Pull request by morozovamv**

### CODE REVIEW

File: packages/react-kit/src/components/DateInput/DateInput.model.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import { Option, getSetoid, some } from 'fp-ts/lib/Option';
+import { setoidNumber, getRecordSetoid } from 'fp-ts/lib/Setoid';
+import { TPopoverProps } from '../Popover/Popover';
+import { TButtonIconProps } from '../ButtonIcon/ButtonIcon';
+import { TControlProps } from '../Control/Control';
+import { TSteppableInputProps } from '../SteppableInput/SteppableInput';
+
+export type TDate = {
```
Comment (raveclassic) [inline packages/react-kit/src/components/DateInput/DateInput.model.ts]: let's name it `TDateInputValue`

File: packages/react-kit/src/components/DateInput/DateInput.model.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	day: Option<number>;
+	month: Option<number>;
+	year: Option<number>;
+};
+
+const setoidOptionNumber = getSetoid(setoidNumber);
+const dateSetoid = getRecordSetoid<TDate>({
```
Comment (raveclassic) [inline packages/react-kit/src/components/DateInput/DateInput.model.ts]: nice ;)

File: packages/react-kit/src/components/DateInput/DateInput.tsx [hunk 3]
```diff
@@ -191,41 +107,61 @@ class RawDateInput extends React.Component<TDateInputFullProps, TDateInputState>
...
+			min,
+			max,
+			theme,
+		} = this.props;
 		const { isOpened } = this.state;
 		if (!Calendar) {
 			return null;
 		}
 
+		const date = sequence(option, array)([day, month, year]).fold(
```
[the remark this answers is not in the document]
Comment (raveclassic) [reply packages/react-kit/src/components/DateInput/DateInput.tsx]: or at least `const buildDataOption = sequenceT(option)`

File: packages/react-kit/src/components/DateInput/DateInput.model.ts (hunk 1 above)
Comment (morozovamv) [reply packages/react-kit/src/components/DateInput/DateInput.model.ts]: done

File: packages/react-kit/src/components/DateInput/DateInput.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (morozovamv) [reply packages/react-kit/src/components/DateInput/DateInput.tsx]: done

**Pull request merged**